app/forms/validators.py

```diff
@@ -223,10 +238,10 @@ def __call__(self, form, field):
 
 
 class DateCheck:
-    def __init__(self, message=None):
+    def __init__(self, message: Optional[str] = None):
         self.message = message or error_messages["INVALID_DATE"]
 
-    def __call__(self, form, field):
+    def __call__(self, form: "QuestionnaireForm", field: DateField) -> None:
```

Comment:
Field for these dates are actually `StringField` instead of our custom date fields, seems to be due to the way the date field is set up.
https://github.com/ONSdigital/eq-questionnaire-runner/blob/master/app/forms/fields/month_year_date_field.py#L9-L12